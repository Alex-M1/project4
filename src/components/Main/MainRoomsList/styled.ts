import styled from 'styled-components';
import { IStyled } from 'common/types/styledTypes';

export const StRooms = styled.div <IStyled>`
    width: 500px;
    height: 90%;
    padding: 10px;
    background: ${({ colors, theme }) => colors[theme].backgroundSecondary};
    color: ${({ colors, theme }) => colors[theme].textColor};
    border-radius: 12px;
`
export const StRoomsContainer = styled.div`
    width: 100%;
    height: 88.5%;
    margin: 10px 0;
    overflow-y: scroll;
`