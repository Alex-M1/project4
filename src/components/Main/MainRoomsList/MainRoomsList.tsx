import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/hooks/useTheme';
import Title from 'common/Title';
import { IRoom } from 'store/room/types';
import { StP, StRooms, StRoomsContainer } from './styled';
import MainRoomsItem from '../MainRoomsItem';
import AddRoomBtn from '../AddRoomBtn';

interface IProps {
    rooms: IRoom[],
    socketConnection: () => void
}

const MainRoomsList: React.FC<IProps> = ({ rooms, socketConnection }) => {
    useEffect(() => {
        socketConnection();
    }, []);
    const { t } = useTranslation();
    const { colors, theme } = useTheme();
    return (
        <StRooms
            theme={theme}
            colors={colors}
        >
            <Title title="list_of_rooms" />
            <AddRoomBtn />
            <StRoomsContainer>
                {rooms.length
                    ? rooms.map((room) => (
                        <MainRoomsItem
                            key={room.id}
                            login={room.creatorLogin}
                            type={room.gameType}
                        />
                    ))
                    : <StP>{t('no_rooms')}</StP>}
            </StRoomsContainer>
        </StRooms>
    );
};

export default MainRoomsList;
