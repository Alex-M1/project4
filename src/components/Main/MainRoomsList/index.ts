import { connect } from 'react-redux';
import { socketConnection } from 'store/room/actions';
import { getRoomList } from 'store/room/selectors';
import { AppStateType } from 'store/rootReducer';
import { getMyOpponentGame } from 'store/ticTac/selectors';
import MainRoomsList from './MainRoomsList';

const mapStateToProps = (state: AppStateType) => ({
  rooms: getRoomList(state),
  myOpponentGame: getMyOpponentGame(state),
});

const mapDispatchToProps = {
  socketConnection,
};

export default connect(mapStateToProps, mapDispatchToProps)(MainRoomsList);
