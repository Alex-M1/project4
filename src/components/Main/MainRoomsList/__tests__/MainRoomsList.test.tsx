import React from 'react';
import { mountSmart, shallowSmart } from 'src/__tests__/testHelper';
import MainRoomsList from '../MainRoomsList';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));

describe('MainRoomsList', () => {
  const props = {
    rooms: [{ id: 'ada', creatorLogin: '', gameType: '' }],
    toRoom: '',
    myOpponentGame: {
      id: 'string',
      guestLogin: 'string',
      startTime: 123123123,
      gameType: 'string',
      stepDtoList: [],
    },
    redirectToRoom: jest.fn(),
    socketConnection: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = shallowSmart(<MainRoomsList {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StRooms', () => {
    const component = mountSmart(<MainRoomsList {...props} />);
    expect(component.find('styled__StRooms')).toHaveLength(1);
  });
  it('should render StRoomsContainer', () => {
    const component = mountSmart(<MainRoomsList {...props} />);
    expect(component.find('styled__StRoomsContainer')).toHaveLength(1);
  });
  it('should render Title', () => {
    const component = mountSmart(<MainRoomsList {...props} />);
    expect(component.find('Title')).toHaveLength(1);
  });
  it('should render AddRoomBtn', () => {
    const component = mountSmart(<MainRoomsList {...props} />);
    expect(component.find('AddRoomBtn')).toHaveLength(1);
  });
  it('should render MainRoomsItem', () => {
    const component = mountSmart(<MainRoomsList {...props} />);
    expect(component.find('MainRoomsItem')).toHaveLength(props.rooms.length);
  });
  it('should render StP', () => {
    const newProps = { ...props, rooms: [] };
    const component = mountSmart(<MainRoomsList {...newProps} />);
    expect(component.find('styled__StP')).toHaveLength(1);
  });
  it('should call socketConnection', () => {
    mountSmart(<MainRoomsList {...props} />);
    expect(props.socketConnection).toHaveBeenCalled();
  });
  it('should call localStorage', () => {
    const spy = jest
      .spyOn(window.localStorage.__proto__, 'setItem');
    mountSmart(<MainRoomsList {...props} />);
    expect(spy).toHaveBeenCalled();
  });
  it('should call localStorage without myOpponentGame.id', () => {
    const spy = jest
      .spyOn(window.localStorage.__proto__, 'setItem');
    const newProps = { ...props, myOpponentGame: { ...props.myOpponentGame, id: null } };
    mountSmart(<MainRoomsList {...newProps} />);
    expect(spy).toHaveBeenCalled();
  });

  it('should render Redirect', () => {
    const newProps = { ...props, toRoom: 'Checkers' };
    const component = mountSmart(<MainRoomsList {...newProps} />);
    expect(component.find('Redirect')).toHaveLength(1);
  });
  it('should render call redirectToRoom', () => {
    const newProps = { ...props, toRoom: 'Checkers' };
    mountSmart(<MainRoomsList {...newProps} />);
    expect(props.redirectToRoom).toHaveBeenCalled();
  });
});
