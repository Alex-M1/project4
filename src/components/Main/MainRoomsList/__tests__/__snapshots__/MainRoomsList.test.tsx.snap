// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainRoomsList Should match snapshot 1`] = `"<div class=\\"styled__StRooms-cxknzw kQcOlJ\\"><h1 class=\\"styled__StTitle-lfcumM bhHMZN\\"></h1><button class=\\"styled__StBtn-kAydTA jyFLiv\\"></button><div class=\\"styled__StRoomsContainer-bYRoxr bxPYMn\\"><a aria-current=\\"page\\" class=\\"active\\" href=\\"/\\"><div class=\\"styled__StItem-jUSzsl DPkny\\"><p></p><span class=\\"styled__StIcon-chPeWD cDIvgb\\"><img src=\\"src/assets/img/play.png\\" alt=\\"\\"/></span><span class=\\"styled__StIcon-chPeWD cDIvgb\\"><img src=\\"undefined\\" alt=\\"\\"/></span></div></a></div></div>"`;
